File: Data Structures & Algorithms/longest-common-prefix/submission-2.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if len(strs) is 0:
            return ""
        for i in range(0,len(strs[0])):
            ch = strs[0][i]

            for j in range(1,len(strs)):
                if i==len(strs[j]) or ch!= strs[j][i]:
                    return strs[0][:i]
        return strs[0]

File: Data Structures & Algorithms/string-encode-and-decode/submission-2.cpp
class Solution {
public:

    string encode(vector<string>& strs) {
        string res = "";
        for(auto str : strs)
        {
            int len = str.size();
            res += to_string(len)+'#'+str;
        }
        return res;

    }

    vector<string> decode(string s) {
        int n = s.size();
        vector<string>res;
        for(int i=0;i<n;)
        {
            int j=i;
            while(s[j]!='#')
            {
                j++;
            }
            int len =stoi(s.substr(i,j-i));
            i=j+1;
            string word = s.substr(i,len);
            res.push_back(word);
            i += len;
        }
        return res;
    }
};
